packages/bsky/src/api/blob-resolver.ts

```diff
@@ -83,20 +83,16 @@ export async function resolveBlob(
   db: Database,
   idResolver: IdResolver,
 ) {
+  const { ref } = db.db.dynamic
   const cidStr = cid.toString()
+
   const [{ pds }, takedown] = await Promise.all([
     idResolver.did.resolveAtprotoData(did), // @TODO cache did info
     db.db
-      .selectFrom('moderation_action_subject_blob')
-      .select('actionId')
-      .innerJoin(
-        'moderation_action',
-        'moderation_action.id',
-        'moderation_action_subject_blob.actionId',
-      )
-      .where('cid', '=', cidStr)
-      .where('action', '=', TAKEDOWN)
-      .where('reversedAt', 'is', null)
+      .selectFrom('moderation_subject_status')
+      .select('id')
+      .where(sql<string>`${ref('blobCids')} @> ${JSON.stringify([cidStr])}`)
+      .where('takendown', 'is', true)
```

Comment:
I am having a little trouble thinking through how blobs are modeled here.  I think the main concern is that we want these lookups to be quick and cheap for the blob resolver.  But we also might consider tracking blob takedowns separately similar to how they are managed with `updateSubjectStatus` and `getSubjectStatus`: it might make it easier to make this lookup quick/cheap, and also might make blobs easier to work with.

Reply:
So, previously, when an action would contain an array of `subjectBlobCids` it would simply create a row for each item in the array where each row references back to the action entry itself. now, instead of storing these in a separate table, we're just dumping them all in this column.

For perf concerns, I am sure a separate table would be faster for lookups but also, I don't think we have many rows with blob references (speaking purely from observing moderators taking actions, we should verify this from prod db perhaps?) so maybe with an index on this array column, we would have fast enough lookups for now? Happy to take your recommendation on this since I don't have a good idea of the scale we're at right now vs. where we will be before this becomes an issue.

on `getSubjectStatus` it looks like we are using `did`, `cid` and `recordUri` which is basically the same as other records/strongRefs except that for those we're not using explicit `did`, right? so, if wanted to start treating blobs as individual subjects, wouldn't we also need some kind of flag column in this table that tells us which entry is for a blob and which one is for a generic record, since both of them would have a `did`, `recordPath` and a `cid` ?